### ISSUE TITLE
Type error

### ISSUE THREAD

**Issue by KilianRK** [CarliJoy/GLS_account_statement_reader#2]
Hi,

I am interested in using this tool for GLS statements. However, when running the command "statement2csv" I get the following error. I am using python 3.7

  _```
File "/usr/local/bin/statement2csv", line 10, in <module>
    sys.exit(run())
  File "/home/xxx/.local/lib/python3.7/site-packages/bank_statement_reader/cli.py", line 57, in run
    main(args or sys.argv[1:])
  File "/home/xxx/.local/lib/python3.7/site-packages/bank_statement_reader/cli.py", line 49, in main
    bookings: Bookings = files2booking(files)
  File "/home/xxx/.local/lib/python3.7/site-packages/bank_statement_reader/statement_reader.py", line 267, in files2booking
    bookings = bookings + pdf2bookings(filename)
  File "/home/xxx/.local/lib/python3.7/site-packages/bank_statement_reader/bookings.py", line 52, in __add__
    for itm in other:
  File "/home/xxx/.local/lib/python3.7/site-packages/bank_statement_reader/bookings.py", line 44, in __iter__
    for itm in sorted(super().__iter__()):
  File "/home/xxx/.local/lib/python3.7/site-packages/bank_statement_reader/booking/booking_base.py", line 292, in __lt__
    return self.date < other.date
TypeError: '<' not supported between instances of 'NoneType' and 'datetime.date
```_

**Comment by CarliJoy** [comment]
Could you send me an example file to `kound@posteo.de` ?

Otherwise it is hard to debug.

**Comment by KilianRK** [comment]
Hi, Thanks for the quick answer. My bank statement is a little bit to sensitive to send you, though. Is there an alternative way?

**Comment by KilianRK** [comment]
Could it be a similar error as in this post?

https://stackoverflow.com/questions/43708541/python-typeerror-typeerror-not-supported-between-instances-of-nonetype

**Comment by CarliJoy** [comment]
Understandable. You could convert the statement to a txt file using `pdftotext -layout bank_report.pdf -` and remove critical stuff from there. But try to not change the length of the strings. I.e. replace name `John Doe` with `Xxxx Xxx` and `123,34` with `111,11`. If you change the length of the strings or remove spaces, things could still brake between the file you give me and the file that is processed.
The issues arises as the script can't determine the correct date and therefore can't sort the entries correctly.

Without a test file there is unfortunately no way to fix it.

**Comment by KilianRK** [comment]
That would be a lot of work. Can I delete most of the statement that you have just the header plus one transfer?

**Comment by CarliJoy** [comment]
@KilianRK learn regex, that could help you alot ;-)
You can try that but make sutre that this file causes the same error

**Comment by KilianRK** [comment]
Okay, I'll try to set up an example file.

**Comment by CarliJoy** [comment]
@KilianRK any updates?

**Comment by ni-ka** [comment]
@CarliJoy I have the same error and happy to generate a test file. So you don't need a pdf but just the text with obsfucated data (while keeping the length)?

**Comment by CarliJoy** [comment]
Exactly. Helpful would also be a screenshot of the PDF Statement, with obfuscated private data. 

I have to note, that I haven't used the tool in a while as the GLS bank now offers CSV exports dating back at least 2 years (1 year in one file max).

Send me a mail with the data if you want.

**Comment by CarliJoy** [comment]
Still can't reproduce due to missing test data.
